.github/workflows/nightly-build.yaml

```diff
@@ -0,0 +1,24 @@
+name: Nightly Build
+
+on:
+  schedule:
+    - cron: "0 0 * * *"
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          ref: master
+
+      - name: Build and push Docker image
+        env:
+          DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+          DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+        run: |
```

Comment:
Before building the nightly, can we also update the version str like we're planning to do in #1446? See reference code here: https://github.com/skypilot-org/skypilot/blob/ee1e57639f84a8143b18e7f8c8af55c4de462ef1/.github/workflows/pypi-nightly-build.yml#L40-L45